src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java

```diff
@@ -117,52 +109,47 @@ static Preferences getUserRoot() {
-        AccessController.doPrivileged(new PrivilegedAction<Void>() {
-            public Void run() {
-                userRootDir =
-                      new File(System.getProperty("java.util.prefs.userRoot",
-                      System.getProperty("user.home")), ".java/.userPrefs");
-                // Attempt to create root dir if it does not yet exist.
-                if (!userRootDir.exists()) {
-                    if (userRootDir.mkdirs()) {
-                        try {
-                            chmod(userRootDir.getCanonicalPath(), USER_RWX);
-                        } catch (IOException e) {
-                            getLogger().warning("Could not change permissions" +
-                                " on userRoot directory. ");
-                        }
-                        getLogger().info("Created user preferences directory.");
-                    }
-                    else
-                        getLogger().warning("Couldn't create user preferences" +
-                        " directory. User preferences are unusable.");
-                }
-                isUserRootWritable = userRootDir.canWrite();
-                String USER_NAME = System.getProperty("user.name");
-                userLockFile = new File (userRootDir,".user.lock." + USER_NAME);
-                userRootModFile = new File (userRootDir,
-                                               ".userRootModFile." + USER_NAME);
-                if (!userRootModFile.exists())
+        userRootDir =
+              new File(System.getProperty("java.util.prefs.userRoot",
+              System.getProperty("user.home")), ".java/.userPrefs");
+        // Attempt to create root dir if it does not yet exist.
+        if (!userRootDir.exists()) {
+            if (userRootDir.mkdirs()) {
                 try {
-                    // create if does not exist.
-                    userRootModFile.createNewFile();
-                    // Only user can read/write userRootModFile.
-                    int result = chmod(userRootModFile.getCanonicalPath(),
-                                                               USER_READ_WRITE);
-                    if (result !=0)
-                        getLogger().warning("Problem creating userRoot " +
-                            "mod file. Chmod failed on " +
-                             userRootModFile.getCanonicalPath() +
-                             " Unix error code " + result);
+                    chmod(userRootDir.getCanonicalPath(), USER_RWX);
                 } catch (IOException e) {
-                    getLogger().warning(e.toString());
+                    getLogger().warning("Could not change permissions" +
+                        " on userRoot directory. ");
                 }
-                userRootModTime = userRootModFile.lastModified();
-                return null;
+                getLogger().info("Created user preferences directory.");
             }
-        });
+            else
```

Comment:
It seems odd that `else` is not on line 125 after the bracket. If putting it on the same line makes it  less readable, then maybe add a pair of brackets for the `else`.